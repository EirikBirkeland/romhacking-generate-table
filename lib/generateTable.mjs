export default function generateTable(startChar, startHex) {
    const alphabet = Array.from(isLowerCase(startChar) ? "abcdefghijklmnopqrstuvxyz" : "ABCDEFGHIJKLMNOPQRSTUVWXYZ");

    return alphabet.map((letter, index) => {
        return [letter, (parseInt(startHex, 16) + index).toString(16).toUpperCase()];
    });

    function isLowerCase(str) {
        return str.toLowerCase() === str;
    }
}