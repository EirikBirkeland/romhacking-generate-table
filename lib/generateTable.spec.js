import generateTable from './generateTable.mjs';

describe('generateTable utility', () => {
    it('creates a table for lowercase', () => {
        const resLower = generateTable("a", 40);

        expect(resLower).toMatchSnapshot();
    });
    it('creates a table for uppercase', () => {
        const resUpper = generateTable("A", 80);

        expect(resUpper).toMatchSnapshot();
    });
});