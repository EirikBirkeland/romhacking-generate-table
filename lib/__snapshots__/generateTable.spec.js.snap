// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTable utility creates a table for lowercase 1`] = `
Array [
  Array [
    "a",
    "40",
  ],
  Array [
    "b",
    "41",
  ],
  Array [
    "c",
    "42",
  ],
  Array [
    "d",
    "43",
  ],
  Array [
    "e",
    "44",
  ],
  Array [
    "f",
    "45",
  ],
  Array [
    "g",
    "46",
  ],
  Array [
    "h",
    "47",
  ],
  Array [
    "i",
    "48",
  ],
  Array [
    "j",
    "49",
  ],
  Array [
    "k",
    "4A",
  ],
  Array [
    "l",
    "4B",
  ],
  Array [
    "m",
    "4C",
  ],
  Array [
    "n",
    "4D",
  ],
  Array [
    "o",
    "4E",
  ],
  Array [
    "p",
    "4F",
  ],
  Array [
    "q",
    "50",
  ],
  Array [
    "r",
    "51",
  ],
  Array [
    "s",
    "52",
  ],
  Array [
    "t",
    "53",
  ],
  Array [
    "u",
    "54",
  ],
  Array [
    "v",
    "55",
  ],
  Array [
    "x",
    "56",
  ],
  Array [
    "y",
    "57",
  ],
  Array [
    "z",
    "58",
  ],
]
`;

exports[`generateTable utility creates a table for uppercase 1`] = `
Array [
  Array [
    "A",
    "80",
  ],
  Array [
    "B",
    "81",
  ],
  Array [
    "C",
    "82",
  ],
  Array [
    "D",
    "83",
  ],
  Array [
    "E",
    "84",
  ],
  Array [
    "F",
    "85",
  ],
  Array [
    "G",
    "86",
  ],
  Array [
    "H",
    "87",
  ],
  Array [
    "I",
    "88",
  ],
  Array [
    "J",
    "89",
  ],
  Array [
    "K",
    "8A",
  ],
  Array [
    "L",
    "8B",
  ],
  Array [
    "M",
    "8C",
  ],
  Array [
    "N",
    "8D",
  ],
  Array [
    "O",
    "8E",
  ],
  Array [
    "P",
    "8F",
  ],
  Array [
    "Q",
    "90",
  ],
  Array [
    "R",
    "91",
  ],
  Array [
    "S",
    "92",
  ],
  Array [
    "T",
    "93",
  ],
  Array [
    "U",
    "94",
  ],
  Array [
    "V",
    "95",
  ],
  Array [
    "W",
    "96",
  ],
  Array [
    "X",
    "97",
  ],
  Array [
    "Y",
    "98",
  ],
  Array [
    "Z",
    "99",
  ],
]
`;
